packages/editor/src/components/block-settings-menu/index.js

```diff
@@ -199,15 +235,32 @@ export default compose( [
 		} );
 
 		return {
-			index: getBlockIndex( last( castArray( clientIds ) ), rootClientId ),
+			firstSelectedIndex: getBlockIndex( first( castArray( clientIds ) ), rootClientId ),
+			lastSelectedIndex: getBlockIndex( last( castArray( clientIds ) ), rootClientId ),
 			isLocked: !! getTemplateLock( rootClientId ),
 			blocks,
 			canDuplicate,
 			shortcuts,
 		};
 	} ),
-	withDispatch( ( dispatch, { clientIds, rootClientId, blocks, index, isLocked, canDuplicate } ) => {
-		const { insertBlocks, multiSelect, removeBlocks, selectBlock } = dispatch( 'core/editor' );
+	withDispatch( ( dispatch, props ) => {
+		const {
+			clientIds,
+			rootClientId,
+			blocks,
+			firstSelectedIndex,
+			lastSelectedIndex,
+			isLocked,
+			canDuplicate,
+		} = props;
```

Comment:
Nit: you could put newlines into these function arguments if you don't want `props` in the scope.

```js
withDispatch( ( dispatch, {
	clientIds,
	rootClientId,
	// etc.
} ) ) {
	// etc.
}
```